### PR TITLE
OpenShift backend

### PR DESCRIPTION
**Pull request by dAnjou**
I just copied the Kubernetes backend. I've tested it on a real OpenShift deployment.

### CODE REVIEW

**Comment by dAnjou** [comment]
Hmm, no idea why this is failing. The tests seem to be completely unrelated 🤔

**Comment by philpep** [comment]
Hi this looks almost good to me, thanks for this.
Can you document usage of the shell parameter ? Don't worry about broken tests, they are unrelated.

File: doc/source/backends.rst [hunk 1]
```diff
@@ -79,16 +79,16 @@ ansible
...
 (or ``--force-ansible``) is set.
 
 For other connections types or when `force_ansible=True`, testinfra will run
 all commands through ansible, which is substantially slower than using native
 connections backends.
 
-If ssh identity file is not provided via `--ssh-identity-file` flag, testinfra will try 
-to use `ansible_ssh_private_key_file`, `ansible_private_key_file` and, finally, 
+If ssh identity file is not provided via `--ssh-identity-file` flag, testinfra will try
+to use `ansible_ssh_private_key_file`, `ansible_private_key_file` and, finally,
```
Comment (philpep) [inline doc/source/backends.rst]: These changes are unrelated, If you feel enough confident with git you can try to put them in a separate commit, and squash openshift related commits in a single commit (otherwise, not a big deal ;) )

**Comment by dAnjou** [comment]
Done. Added tests for the shell feature as well.

**Comment by philpep** [comment]
Merged, thanks!
PS: I'm not sure I'll keep the custom shell implementation as it, let's continue the discussion on #516

**Comment by dAnjou** [comment]
Thanks, looking forward to the next release!
